dbt/models/default/default.vw_pin_sale.sql

```diff
@@ -249,93 +259,198 @@ sales_val AS (
+        COALESCE(uq_sales.sale_price, md_sales.sale_price)
+            AS sale_price_coalesced, --noqa
+        uq_sales.sale_key,
+        COALESCE(uq_sales.doc_no, md_sales.doc_no) AS doc_no_coalesced, --noqa
+        COALESCE(uq_sales.deed_type, md_sales.mydec_deed_type)
+            AS deed_type_coalesced,
+        uq_sales.deed_type AS deed_type_ias,
+        COALESCE(uq_sales.seller_name, md_sales.seller_name)
+            AS seller_name_coalesced,
+        COALESCE(uq_sales.is_multisale, md_sales.is_multisale)
+            AS is_multisale_coalesced,
+        COALESCE(uq_sales.num_parcels_sale, md_sales.num_parcels_sale)
+            AS num_parcels_sale_coalesced,
+        COALESCE(uq_sales.buyer_name, md_sales.buyer_name)
+            AS buyer_name_coalesced,
+        COALESCE(uq_sales.sale_type, NULL) AS sale_type_coalesced,
+        uq_sales.max_price,
+        uq_sales.bad_doc_no,
+        CASE WHEN uq_sales.doc_no IS NOT NULL THEN 'iasworld' ELSE 'mydec' END
+            AS source,
+        md_sales.mydec_deed_type,
+        md_sales.sale_filter_ptax_flag,
+        md_sales.mydec_property_advertised,
+        md_sales.mydec_is_installment_contract_fulfilled,
+        md_sales.mydec_is_sale_between_related_individuals_or_corporate_affiliates, --noqa
+        md_sales.mydec_is_transfer_of_less_than_100_percent_interest,
+        md_sales.mydec_is_court_ordered_sale,
+        md_sales.mydec_is_sale_in_lieu_of_foreclosure,
+        md_sales.mydec_is_condemnation,
+        md_sales.mydec_is_short_sale,
+        md_sales.mydec_is_bank_reo_real_estate_owned,
+        md_sales.mydec_is_auction_sale,
+        md_sales.mydec_is_seller_buyer_a_relocation_company,
+        md_sales.mydec_is_seller_buyer_a_financial_institution_or_government_agency, --noqa
+        md_sales.mydec_is_buyer_a_real_estate_investment_trust,
+        md_sales.mydec_is_buyer_a_pension_fund,
+        md_sales.mydec_is_buyer_an_adjacent_property_owner,
+        md_sales.mydec_is_buyer_exercising_an_option_to_purchase,
+        md_sales.mydec_is_simultaneous_trade_of_property,
+        md_sales.mydec_is_sale_leaseback,
+        md_sales.mydec_is_homestead_exemption,
+        md_sales.mydec_homestead_exemption_general_alternative,
+        md_sales.mydec_homestead_exemption_senior_citizens,
+        md_sales.mydec_homestead_exemption_senior_citizens_assessment_freeze
+    FROM unique_sales AS uq_sales
+    -- This logic brings in mydec sales that aren't in iasworld.
+    -- If a doc_no exists in iasworld and mydec, we prioritize iasworld,
+    -- if it only exists in mydec, we will grab the doc_no from mydec. The
+    -- 'source' column lets us know which table the doc_no came from and allows
+    -- us to filter for only iasworld sales or for mydec sales that aren't in
+    -- iasworld already.
+    FULL OUTER JOIN mydec_sales AS md_sales
+        ON uq_sales.doc_no = md_sales.doc_no
+        AND md_sales.is_multisale = FALSE
```

Comment:
> So you're right about how the view has worked in the past, but this is ultimately still something we're going to have to address. We didn't let mydec multisales in because we were only joining mydec sales to iasworld sales and that would have potentially been a one to many join on sale doc in some cases. But now that mydec sales are in the view in their own right, we'll need to figure out a way to allow in those multisales, without joining them to iasworld sales. I think you can probably swing this with a couple of extra conditionals in FULL OUTER JOIN mydec_sales AS md_sales ON uq_sales.doc_no = md_sales.doc_no to make sure multisale mydec sales don't join onto iasworld sales but can exist if they don't have a matching iasworld sale.

Does this fix our problem of not letting multi-sale mydec date data join and override the iasworld multi-sale date? I think it at least mostly does:

```sql 
select count(*), is_mydec_date
from "z_ci_583_create_a_sales_view_that_combines_iasworld_mydec_and_ccrd_sales_default"."vw_pin_sale"
where source = 'iasworld'
group by is_mydec_date

- - -
#	_col0	is_mydec_date
1	1582983	false
2	913090	true
```
```sql 
select count(*), is_mydec_date
from "default"."vw_pin_sale"
group by is_mydec_date

- - -

#	_col0	is_mydec_date
1	1582903	false
2	913170	true
```

Reply:
i think you're gonna have to be a little bit more precise with the join. it'll have to be something like
```
FULL OUTER JOIN mydec_sales AS md_sales
        ON (uq_sales.doc_no = md_sales.doc_no AND md_sales.is_multisale = FALSE)
        OR uq_sales.doc_no != md_sales.doc_no
```

Reply:
I feel like this should do the trick, but i'm seeing a lot of small disagreements across columns in skim, now

Reply:
I think 
```
WHERE num_single_day_sales = 1
        OR year > '2020'
```
is causing the discrepancy. doc number 1607434035 seems to be an example of a mydec sale that is being excluded when it shouldn't be.

Reply:
Aside from these, which other columns were you looking at?

```R
skim <- bind_rows(old, new_filtered) %>%
  select(
    new,
    year,
    sale_filter_same_sale_within_365,
    sale_filter_less_than_10k,
    sale_filter_deed_type,
    sale_filter_is_outlier
  ) %>%
  group_by(new) %>%
  skim()
```

Reply:
Stuff like `seller_name`, `mydec_deed_type`. They're pretty small differences.

Reply:
1607434035 presents a very awkward problem.